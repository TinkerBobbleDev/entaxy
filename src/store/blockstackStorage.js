/* eslint-disable no-console */
import { getFile, putFile, isUserSignedIn } from 'blockstack'

const filename = 'entaxy.json'

export const loadState = () => {
  if (isUserSignedIn()) {
    return getFile(filename)
      .then(data => JSON.parse(data))
      .catch((error) => {
        console.log(`Error loading file ${filename}:`, error)
        return {}
      })
  }
  return Promise.resolve({})
}

export const saveState = (state) => {
  if (isUserSignedIn()) {
    return putFile(filename, JSON.stringify(state))
  }
  return undefined
}
