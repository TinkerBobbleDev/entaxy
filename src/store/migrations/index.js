import migration0 from './0'
import migration1 from './1'
import migration2 from './2'
import migration3 from './3'

export default {
  0: migration0,
  1: migration1,
  2: migration2,
  3: migration3
}
