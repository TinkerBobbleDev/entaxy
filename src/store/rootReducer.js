import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import settingsReducer from './settings/reducer'
import transactionsReducer from './transactions/reducer'
import marketValuesReducer from './marketValues/reducer'

export default combineReducers({
  user: userReducer,
  settings: settingsReducer,
  transactions: transactionsReducer,
  marketValues: marketValuesReducer
})
