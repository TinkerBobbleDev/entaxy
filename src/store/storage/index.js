/* eslint-disable import/no-cycle */
import { snakeCase } from 'lodash'
import * as blockstackStorage from './blockstackStorage'
import * as localStorage from './localStorage'
import { showOverlay, hideOverlay } from '../settings/actions'
import { initialState as settingsInitialState } from '../settings/reducer'
import { initialState as accountsInitialState } from '../accounts/reducer'
import { initialState as transactionsInitialState } from '../transactions/reducer'
import { initialState as exchangeRatesInitialState } from '../exchangeRates/reducer'
import { initialState as budgetInitialState } from '../budget/reducer'

export const loadItem = (itemName, newState) => (dispatch, getState) => {
  let payload = newState
  if (itemName === 'settings') {
    const { settings } = getState()
    if (settings !== null && settings !== undefined) {
      payload = { ...newState, overlayMessage: settings.overlayMessage }
    }
  }
  dispatch({
    type: `LOAD_${snakeCase(itemName).toUpperCase()}`,
    payload
  })
}

export const loadAllItems = (newState) => (dispatch) => {
  [
    'settings',
    'accounts',
    'transactions',
    'exchangeRates',
    'budget'
  ].map((itemName) => dispatch(loadItem(itemName, newState[itemName] || {})))
}


export const loadState = () => async (dispatch, getState) => {
  if (getState().user === undefined) return {}
  let newState
  switch (getState().user.isAuthenticatedWith) {
    case 'guest':
      dispatch(showOverlay('Loading data from localStorage...'))
      newState = await localStorage.loadState()
      break
    case 'blockstack':
      dispatch(showOverlay('Loading data from Blockstack...'))
      newState = await blockstackStorage.loadState()
      break
    // no default
  }
  if (newState === undefined) newState = { settings: settingsInitialState }
  dispatch(loadAllItems({
    ...newState,
    settings: {
      ...newState.settings,
      snackbarMessage: null
    }
  }))
  return dispatch(hideOverlay())
}

export const resetState = () => (dispatch, getState) => {
  const { settings } = getState()

  // Preserve the existing currency and locale
  dispatch(loadAllItems({
    settings: {
      ...settingsInitialState,
      currency: settings.currency,
      locale: settings.locale
    },
    accounts: accountsInitialState,
    transactions: transactionsInitialState,
    exchangeRates: exchangeRatesInitialState,
    budget: budgetInitialState
  }))
}

export const saveState = () => (_, getState) => {
  let { user } = getState()
  if (user === undefined) user = {}
  switch (user.isAuthenticatedWith) {
    case 'guest':
      return localStorage.saveState(getState())
    case 'blockstack':
      return blockstackStorage.saveState(getState())
    default:
      return undefined
  }
}
