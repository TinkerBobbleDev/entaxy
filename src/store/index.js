import { createStore, applyMiddleware } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunkMiddleware from 'redux-thunk'
import rootReducer from './rootReducer'

const middlewares = [thunkMiddleware]

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger')
  middlewares.push(logger)
}

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middlewares))
)

if (module.hot) {
  module.hot.accept(rootReducer, () => {
    store.replaceReducer(rootReducer)
  })
}

export default store
