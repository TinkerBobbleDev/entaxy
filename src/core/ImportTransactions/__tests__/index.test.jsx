import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../../store'
import { ImportTransactionsComponent } from '../'

describe('Import Transactions', () => {
  const props = {
    classes: { }
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    const component = renderer.create((
      <Provider store={store}>
        <BrowserRouter>
          <ImportTransactionsComponent
            classes={{ ...props }}
          />
        </BrowserRouter>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
