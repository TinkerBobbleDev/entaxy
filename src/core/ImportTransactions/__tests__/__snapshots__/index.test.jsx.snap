// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Transactions matches snapshot 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-9"
    >
      <div
        className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 Header-toolbar-1"
      >
        <div
          className="Logo-logo-42"
        >
          <img
            alt="Entaxy logo"
            className="Logo-image-43"
            src="logo.png"
          />
          <a
            className="MuiTypography-root-45 MuiTypography-h6-62 MuiTypography-colorInherit-74 Logo-title-44"
            href="/"
            onClick={[Function]}
          >
            Entaxy
          </a>
        </div>
        <div>
          <a
            aria-current={null}
            className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
            href="/portfolio"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-81"
            >
              Portfolio
            </span>
            <span
              className="MuiTouchRipple-root-109"
            />
          </a>
          <a
            aria-current={null}
            className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
            href="/transactions"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-81"
            >
              Transactions
            </span>
            <span
              className="MuiTouchRipple-root-109"
            />
          </a>
          <a
            aria-current={null}
            className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-sizeSmall-103 MuiButton-colorInherit-101"
            href="/import-transactions"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="MuiButton-label-81"
            >
              Import Transactions
            </span>
            <span
              className="MuiTouchRipple-root-109"
            />
          </a>
        </div>
        <button
          className="MuiButtonBase-root-106 MuiButton-root-80 MuiButton-text-82 MuiButton-flat-85 MuiButton-colorInherit-101"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-81"
          >
            Login
          </span>
          <span
            className="MuiTouchRipple-root-109"
          />
        </button>
      </div>
    </header>
    <h5
      className="MuiTypography-root-45 MuiTypography-h5-61 MuiTypography-alignCenter-68"
    >
      Import transactions
    </h5>
    <div>
      <p
        className="MuiTypography-root-45 MuiTypography-body1-54 MuiTypography-alignCenter-68"
      >
        Select an institution
      </p>
      <div>
        <div>
          <div
            className="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 MuiCard-root-117"
          >
            <div
              className="MuiCardHeader-root-118 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-121"
              >
                <span
                  className="MuiTypography-root-45 MuiTypography-headline-50 MuiCardHeader-title-122"
                >
                  RBC
                </span>
                <span
                  className="MuiTypography-root-45 MuiTypography-body1-54 MuiTypography-colorTextSecondary-78 MuiCardHeader-subheader-123"
                >
                  Royal Bank
                </span>
              </div>
              <div
                className="MuiCardHeader-action-120"
              >
                <button
                  className="MuiButtonBase-root-106 MuiIconButton-root-124"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-129"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-130"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-109"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 MuiCard-root-117"
          >
            <div
              className="MuiCardHeader-root-118 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-121"
              >
                <span
                  className="MuiTypography-root-45 MuiTypography-headline-50 MuiCardHeader-title-122"
                >
                  BMO
                </span>
                <span
                  className="MuiTypography-root-45 MuiTypography-body1-54 MuiTypography-colorTextSecondary-78 MuiCardHeader-subheader-123"
                >
                  Bank of Montreal
                </span>
              </div>
              <div
                className="MuiCardHeader-action-120"
              >
                <button
                  className="MuiButtonBase-root-106 MuiIconButton-root-124"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-129"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-130"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-109"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 MuiCard-root-117"
          >
            <div
              className="MuiCardHeader-root-118 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-121"
              >
                <span
                  className="MuiTypography-root-45 MuiTypography-headline-50 MuiCardHeader-title-122"
                >
                  TD
                </span>
                <span
                  className="MuiTypography-root-45 MuiTypography-body1-54 MuiTypography-colorTextSecondary-78 MuiCardHeader-subheader-123"
                >
                  TD Canada Trust
                </span>
              </div>
              <div
                className="MuiCardHeader-action-120"
              >
                <button
                  className="MuiButtonBase-root-106 MuiIconButton-root-124"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-129"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-130"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-109"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiPaper-root-11 MuiPaper-elevation1-14 MuiPaper-rounded-12 MuiCard-root-117"
          >
            <div
              className="MuiCardHeader-root-118 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-121"
              >
                <span
                  className="MuiTypography-root-45 MuiTypography-headline-50 MuiCardHeader-title-122"
                >
                  Tangerine
                </span>
                <span
                  className="MuiTypography-root-45 MuiTypography-body1-54 MuiTypography-colorTextSecondary-78 MuiCardHeader-subheader-123"
                >
                  Personal Banking
                </span>
              </div>
              <div
                className="MuiCardHeader-action-120"
              >
                <button
                  className="MuiButtonBase-root-106 MuiIconButton-root-124"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-129"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-130"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-109"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
